http4s/client/src/main/scala/endpoints4s/http4s/client/ChunkedEntities.scala

```diff
@@ -0,0 +1,73 @@
+package endpoints4s.http4s.client
+
+import org.http4s.{Request => Http4sRequest, Response => Http4sResponse}
+import fs2.Chunk
+
+import fs2.Stream
+import cats.syntax.all._
+import endpoints4s.Valid
+import endpoints4s.Invalid
+
+trait ChunkedEntities extends endpoints4s.algebra.ChunkedEntities with EndpointsWithCustomErrors {
+
+  //#stream-type
+  type Chunks[A] = fs2.Stream[Effect, A]
+  //#stream-type
+
+  override def textChunksRequest: (Chunks[String], Http4sRequest[Effect]) => Http4sRequest[Effect] =
+    (stream, req) => req.withEntity(stream)
+
+  override def textChunksResponse: Http4sResponse[Effect] => Effect[Chunks[String]] =
+    res => effect.pure(res.bodyText)
+
+  override def bytesChunksRequest
+      : (Chunks[Array[Byte]], Http4sRequest[Effect]) => Http4sRequest[Effect] =
+    (stream, req) =>
+      req.withBodyStream(
+        stream.flatMap(arr => fs2.Stream.chunk(Chunk.array(arr)))
+      )
+
+  override def bytesChunksResponse: Http4sResponse[Effect] => Effect[Chunks[Array[Byte]]] =
+    res => effect.pure(res.body.chunks.map(_.toArray))
+
+}
+
+trait ChunkedJsonEntities
+    extends endpoints4s.algebra.ChunkedJsonEntities
+    with ChunkedEntities
+    with JsonEntitiesFromCodecs {
+
+  def jsonChunksRequest[A](implicit
+      codec: JsonCodec[A]
+  ): RequestEntity[Chunks[A]] = { (stream, req) =>
+    val encoder = stringCodec(codec)
+    req.withEntity(stream.map(encoder.encode))
+  }
+
+  def jsonChunksResponse[A](implicit
+      codec: JsonCodec[A]
+  ): ResponseEntity[Chunks[A]] = { response =>
+    val decoder = stringCodec[A](codec)
+
+    //I seem to have to force the entire stream here or I'll just get empty results, I can't quite put my finger where it happens. but i suspect if we propagate the stream wrong here the underlying http4s thing is already gone
+    // I suspect it's because of the use of `client.run(req).use(...) in def endpoint in the Endpoints.scala. Nothing forces compilation of the stream until that use is over, thus it gets closed and the underlying stream ends up empty.
```

Comment:
Why do we have to do this change only here, and not in `textChunksResponse`?

Reply:
If I read the tests correctly, which I'm still bad at, I think it's just not tested. I'll get on that.

Reply:
added a couple of tests, but with the new design that exposes the resource we no longer need to force the streams in the first place.